### PR TITLE
libsecp256k1 updated to latest upstream

### PR DESCRIPTION
**Pull request by arkpar**

### CODE REVIEW

**Comment by arkpar** [comment]
Should be merged together:
https://github.com/ethereum/libweb3core/pull/15
https://github.com/ethereum/webthree-helpers/pull/36
https://github.com/ethereum/libethereum/pull/36

File: utils/secp256k1/ext.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file ext.c
+ * Ethereum extensions to libecp256k1
```
Comment (chfast) [inline utils/secp256k1/ext.c:18]: To what?

File: utils/secp256k1/ext.c [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * @authors:
+ *   Arkadiy Paronyan <i@gavwood.com>
```
Comment (chfast) [inline utils/secp256k1/ext.c:20]: email.

File: utils/secp256k1/ext.c [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * @date 2014
```
Comment (chfast) [inline utils/secp256k1/ext.c:21]: It's 2015 already.

File: utils/secp256k1/ext.c [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ */
+
+#include "src/secp256k1.c"
+
+int secp256k1_ecdh_raw(const secp256k1_context* ctx, unsigned char *result, const secp256k1_pubkey *point, const unsigned char *scalar)
```
Comment (chfast) [inline utils/secp256k1/ext.c:26]: Can't it be implemented outside the library, using public API?

File: utils/secp256k1/CMakeLists.txt [hunk 5]
```diff
@@ -1,8 +1,10 @@
 
-file(GLOB HEADERS "*.h")
+file(GLOB HEADERS "src/*.h")
 
-add_library(secp256k1 secp256k1.c include/secp256k1.h ${HEADERS})
+add_library(secp256k1 ext.c include/secp256k1.h include/ext.h ${HEADERS})
 target_compile_definitions(secp256k1 INTERFACE ETH_HAVE_SECP256K1 PRIVATE HAVE_CONFIG_H)
+target_include_directories(secp256k1 PUBLIC .)
```
Comment (chfast) [inline utils/secp256k1/CMakeLists.txt:6]: That's quite suspicious. Why is secp256k1/include not enough for PUBLIC includes?

Either way you can place in one target_include_directories call.

**Comment by chfast** [comment]
What is the revision of the library?

**Comment by arkpar** [comment]
Closing this per @subtly's request

**Comment by chfast** [comment]
Why?

**Pull request closed without merging**